### PR TITLE
5843.4 Implement Dependency Injection for block and replace old code with services methods.

### PR DESCRIPTION
**Pull request by bobik-dev-branck**
[…e with services methods.](https://redmine.devbranch.work/issues/5843)

### CODE REVIEW

File: web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php [hunk 1]
```diff
@@ -12,29 +15,73 @@
  *   admin_label = @Translation("Exchange_rates"),
  * )
  */
-class ExchangeRatesBlock extends BlockBase {
+class ExchangeRatesBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Constructor.
```
Comment (Y0r) [inline web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:21]: The constructor of what?

File: web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php [hunk 2]
```diff
@@ -12,29 +15,73 @@
...
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   Container.
+   * @param array $configuration
+   *   Configuration array.
+   * @param string $plugin_id
+   *   Plugin ID.
+   * @param mixed $plugin_definition
+   *   Plugin definition.
+   */
```
Comment (Y0r) [inline web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:48]: actually its useless, but there no problems

File: web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php [hunk 3]
```diff
@@ -12,29 +15,73 @@
...
-    $date = date("Ymd");
-    $response = $httpClient
-      ->get('https://bank.gov.ua/NBUStatService/v1/statdirectory/exchange?date=' . $date . '&json')
-      ->getBody();
-    $exchangeRates = json_decode($response);
-
-    foreach ($exchangeRates as $exchangeRate) {
-      $data[$exchangeRate->cc] = $exchangeRate->rate;
+    $showBlock = $this->exchangeRates->getConfig('show_block');
+    if ($showBlock == FALSE) {
```
Comment (Y0r) [inline web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:63]: !$showBlock

File: web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php [hunk 4]
```diff
@@ -12,29 +15,73 @@
...
+      return;
     }
 
-    $renderable = [
-      '#theme' => 'block_exchange_rates',
-      '#data' => $data,
-    ];
+    if ($showBlock == TRUE) {
```
Comment (Y0r) [inline web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:67]: $showBlock

**Review by Y0r — APPROVED** [review]
Solve problems before merge

File: web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php (hunk 1 above)
Comment (bobik-dev-branck) [reply web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:21]: Fixed

Comment (bobik-dev-branck) [reply web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:21]: Fixed

File: web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php (hunk 2 above)
Comment (bobik-dev-branck) [reply web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:48]: Fixed

File: web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php (hunk 3 above)
Comment (bobik-dev-branck) [reply web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:63]: Fixed

File: web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php (hunk 4 above)
Comment (bobik-dev-branck) [reply web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:67]: Fixed

**Pull request merged**